x/assets/keeper/operator_asset.go

```diff
@@ -10,27 +10,27 @@ import (
 
 // This file provides all functions about operator assets state management.
 
-func (k Keeper) GetOperatorAssetInfos(ctx sdk.Context, operatorAddr sdk.Address, _ map[string]interface{}) (assetsInfo map[string]*assetstype.OperatorAssetInfo, err error) {
-	store := prefix.NewStore(ctx.KVStore(k.storeKey), assetstype.KeyPrefixOperatorAssetInfos)
-	// the key is the operator address in the bech32 format
-	key := []byte(operatorAddr.String())
-	iterator := sdk.KVStorePrefixIterator(store, key)
-	defer iterator.Close()
-
+func (k Keeper) GetOperatorAssetInfos(ctx sdk.Context, operatorAddr sdk.Address, assetsFilter map[string]interface{}) (assetsInfo map[string]*assetstype.OperatorAssetInfo, err error) {
 	ret := make(map[string]*assetstype.OperatorAssetInfo, 0)
-	for ; iterator.Valid(); iterator.Next() {
-		var stateInfo assetstype.OperatorAssetInfo
-		k.cdc.MustUnmarshal(iterator.Value(), &stateInfo)
-		keyList, err := assetstype.ParseJoinedStoreKey(iterator.Key(), 2)
-		if err != nil {
-			return nil, err
+	opFunc := func(assetID string, state *assetstype.OperatorAssetInfo) error {
+		if _, ok := assetsFilter[assetID]; ok {
+			ret[assetID] = state
```

Comment:
1. `IteratorAssetsForOperator` has checked `assetsFilter`, so opFunc may not need to do it again (maybe better just keep the filter in `opFunc`)
2. in `Itera...`, if the assetsFilter is nil, that means 'all assets', but here will skip all(and also will cause panic)

Reply:
Yeah, thanks. I will fix it.